File: jeux/loup-garou/game.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>
#include "game.h"

void initPlayers(Player players[], int numPlayers) {
    for(int i=0;i<numPlayers;i++){
        printf("Entrez le nom du joueur %d: ", i+1);
        scanf("%s", players[i].name);
        players[i].alive = 1;
        // Attribution simple: 1 loup, 1 voyante, 1 chasseur, le reste villageois
        if(i==0) players[i].role = LOUP;
        else if(i==1) players[i].role = VOYANTE;
        else if(i==2) players[i].role = CHASSEUR;
        else players[i].role = VILLAGEOIS;
    }
}

File: jeux/loup-garou/game.py
import pygame
from settings import *
from roles import *

class Game:
    def __init__(self):
        pygame.init()
        self.screen = pygame.display.set_mode((WIDTH, HEIGHT))
        pygame.display.set_caption("Loup Garou")
        self.clock = pygame.time.Clock()

        # Exemple de joueurs + rôles
        self.players = [
            {"name": "Alice", "role": LoupGarou()},
            {"name": "Bob", "role": Voyante()},
            {"name": "Claire", "role": Villageois()}
        ]

        self.font = pygame.font.SysFont("arial", 28)

    def draw_text(self, text, x, y):
        img = self.font.render(text, True, WHITE)
        self.screen.blit(img, (x, y))

    def run(self):
        running = True
        while running:
            self.clock.tick(FPS)

            for event in pygame.event.get():
                if event.type == pygame.QUIT:
                    running = False

            self.screen.fill(BLACK)

            # Affichage simple des joueurs et rôles
            y = 50
            for p in self.players:
                self.draw_text(f"{p['name']} : {p['role'].name}", 50, y)
                y += 40

            pygame.display.flip()

        pygame.quit()
